### DISCUSSION TITLE
How to debug JS hook scripts?

### DISCUSSION THREAD

**Discussion by Creling** [pocketbase/pocketbase#3870] [Q&A]
Hi,

It might be a silly question, but, how can I debug the js hook scripts? I mean, if I console.log somethings, where could I see the outputs?

Creling

**Comment by ganigeorgiev** [comment]
> if I console.log somethings, where could I see the outputs?

The output should be visible in the terminal where the PocketBase executable was started.

The `console` JS hooks module works in a similar manner as in Node.js:
- `console.log`, `console.info`, `console.debug` prints to stdout
- `console.error`, `console.warn` prints to stderr

With the next v0.20.0 release non-request hook errors will be also available in the _Admin UI > Logs_ panel.

**Reply by hfichter** [reply]
Any chance to attach a JS debugger? I assume not, as the doc only mentioned logs, but couldn't find a statement on that.
I'm fairly new to JS ecosystem, and I don't know Go so please indulge me. 
Great software by the way! ❤️
